### PR TITLE
[examples] Add deliveroo.temple

### PR DESCRIPTION
**Pull request by Lewiky**

### CODE REVIEW

File: examples/deliveroo.temple [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+// A dish is made by a restaurant and is delivered to a user
+Dish: service {
+    name: string;
+    description: string;
+    price: float(0.0);
+    restaurant: Restaurant;
+
+    Img: struct {
+        dish: Dish;
```
Comment (jaylees14) [inline examples/deliveroo.temple]: I think this reference to the parent is implicit, so you don't need it

**Review by jaylees14 — APPROVED** [review]
Awesome, just one point

File: examples/deliveroo.temple [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        img: data(4M);
+        #enumerable;
+    }
+    #enumerable;
+    #writable(by: this);
+    // Implcit #readable(by: all)
```
Comment (c-harding) [inline examples/deliveroo.temple]:
```suggestion
    // Implicit #readable(by: all)
```

File: examples/deliveroo.temple [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+// Users order dishes, paid for with their billing account
+User: service {
+    firstName: string;
+    lastName: string;
+    email: string;
+    address: string;
+    billingAccount: Billing;
+    #auth;
+    // Implicit #readable(by:this)
```
Comment (c-harding) [inline examples/deliveroo.temple:18]:
```suggestion
    // Implicit #readable(by: this)
```

File: examples/deliveroo.temple [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    // Implicit #writable(by: this)
+}
+
+// Riders deliver orders - location so user knows where they are
+Rider: service {
+    firstName: string;
+    lat: float(85.0, -85.0);
```
Comment (c-harding) [inline examples/deliveroo.temple]:
```suggestion
    lat: float(90.0, -90.0);
```

File: examples/deliveroo.temple [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    restaurant: Restaurant;
+    rider: Rider;
+    dish: Dish;
+    address: string;
+    orderTime: datetime @serverSet;
+    status: string;
+    #enumerable;
+    #writable(by:this);
+    #omit[delete];
+    // Implcit #readable(by:all)
```
Comment (c-harding) [inline examples/deliveroo.temple]:
```suggestion
    // Implicit #readable(by:all)
```

File: examples/deliveroo.temple [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    #readable(by: all);
+    // Implicit #writable(by: this)
+}
+
+// An order marks a dish to be delivered by a rider from a restaurant to a user
+Order: service {
+    user: User;
+    restaurant: Restaurant;
+    rider: Rider;
+    dish: Dish;
```
Comment (c-harding) [inline examples/deliveroo.temple]: just one? seems like the perfect use case for a struct acting as an array

**Review by c-harding — APPROVED** [review]

**Pull request merged**